lib/Jsrt/JsrtHelper.cpp

```diff
@@ -89,7 +89,7 @@ void JsrtCallbackState::ObjectBeforeCallectCallbackWrapper(JsObjectBeforeCollect
         pthread_key_create(&s_threadLocalDummy, DISPOSE_CHAKRA_CORE_THREAD);
 #endif
 
-#if defined(CHAKRA_STATIC_LIBRARY)
+#if defined(CHAKRA_STATIC_LIBRARY) || !defined(_WIN32)
```

Comment:
> Actually by enabling this part; you are using the static lib interface for shared lib. So, it is interesting that it doesn't hang.

From the code, you had following effective on static lib *and* xplat shared lib:
```
#if defined(CHAKRA_STATIC_LIBRARY) || !defined(_WIN32)
_NOINLINE void VALIDATE_ENTER_CURRENT_THREAD();
```
VALIDATE_ENTER_CURRENT_THREAD ensures 2 things lazily -- InitProcess and InitThread. Previously InitThread is done for xplat shared lib -- not sure if you did that intentionally or accidentally. Any way, this change turns on InitProcess, and it works...



> Please feel free to enable this trick until I introduce lazy init to PAL though. I really appreciate if you can add a todo note to me there.

Sure